pkg/storageincentives/agent.go

```diff
@@ -254,28 +249,33 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 
 		// write the current phase only once
-
 		if currentPhase != prevPhase {
 
 			a.metrics.CurrentPhase.Set(float64(currentPhase))
 
 			a.logger.Info("entering phase", "phase", currentPhase.String(), "round", round, "block", block)
 
+			a.nodeState.SetPhase(currentPhase)
+			a.nodeState.SetRound(round)
+			a.nodeState.SetBlock(block)
+			a.nodeState.SaveStatus()
+
```

Comment:
Since you only use the SetPhase, SetRound, and SetBlock here, 
you can have a new single func that takes the three values and saves the status internally